### ISSUE TITLE
geom_sf() not rounding x-axis ticks

### ISSUE THREAD

**Issue by jonasViehweger** [r-spatial/sf#1526]
I have an issue plotting an sf object in ETRS89 / UTM zone 32N. It is not displaying the x-axis coordinate ticks well. Instead of 8.649° it is displaying 8.6490000000000000001°. Also see the image below for how it looks to me.

![grafik](https://user-images.githubusercontent.com/53001455/98236837-446b3780-1f64-11eb-9c6c-acbbeeece4e3.png)

Here's the (hopefully) reproducible code:

    test <- structure(list(structure(list(structure(c(477799.11, 477832.22, 
                                              477832.22, 477799.11, 477799.11, 5632217.3, 5632217.3, 5632250.43, 
                                              5632250.43, 5632217.3), .Dim = c(5L, 2L))), class = c("XY", "POLYGON", 
                                                                                                    "sfg"))), class = c("sfc_POLYGON", "sfc"), precision = 0, bbox = structure(c(xmin = 477799.11, 
                                                                                                                                                                                 ymin = 5632217.3, xmax = 477832.22, ymax = 5632250.43), class = "bbox"), crs = structure(list(
                                                                                                                                                                                   input = "ETRS89 / UTM zone 32N", wkt = "PROJCRS[\"ETRS89 / UTM zone 32N\",\n    BASEGEOGCRS[\"ETRS89\",\n        DATUM[\"European Terrestrial Reference System 1989\",\n            ELLIPSOID[\"GRS 1980\",6378137,298.257222101,\n                LENGTHUNIT[\"metre\",1]]],\n        PRIMEM[\"Greenwich\",0,\n            ANGLEUNIT[\"degree\",0.0174532925199433]],\n        ID[\"EPSG\",4258]],\n    CONVERSION[\"UTM zone 32N\",\n        METHOD[\"Transverse Mercator\",\n            ID[\"EPSG\",9807]],\n        PARAMETER[\"Latitude of natural origin\",0,\n            ANGLEUNIT[\"degree\",0.0174532925199433],\n            ID[\"EPSG\",8801]],\n        PARAMETER[\"Longitude of natural origin\",9,\n            ANGLEUNIT[\"degree\",0.0174532925199433],\n            ID[\"EPSG\",8802]],\n        PARAMETER[\"Scale factor at natural origin\",0.9996,\n            SCALEUNIT[\"unity\",1],\n            ID[\"EPSG\",8805]],\n        PARAMETER[\"False easting\",500000,\n            LENGTHUNIT[\"metre\",1],\n            ID[\"EPSG\",8806]],\n        PARAMETER[\"False northing\",0,\n            LENGTHUNIT[\"metre\",1],\n            ID[\"EPSG\",8807]]],\n    CS[Cartesian,2],\n        AXIS[\"(E)\",east,\n            ORDER[1],\n            LENGTHUNIT[\"metre\",1]],\n        AXIS[\"(N)\",north,\n            ORDER[2],\n            LENGTHUNIT[\"metre\",1]],\n    USAGE[\n        SCOPE[\"unknown\"],\n        AREA[\"Europe - 6Â°E to 12Â°E and ETRS89 by country\"],\n        BBOX[38.76,6,83.92,12]],\n    ID[\"EPSG\",25832]]"), class = "crs"), n_empty = 0L)
    
    ggplot() +
      geom_sf(data = test)

Interestingly using a slightly wider `xlim` fixes it:

```
ggplot() +
  geom_sf(data = test) +
  coord_sf(xlim = c(477796, 477835))
```

Here's my sessionInfo and extSoftVersion

```
          GEOS           GDAL         proj.4 GDAL_with_GEOS     USE_PROJ_H 
       "3.8.0"        "3.0.4"        "6.3.1"         "true"         "true" 
```

```
R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19041)

Matrix products: default

locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252    LC_MONETARY=German_Germany.1252
[4] LC_NUMERIC=C                    LC_TIME=German_Germany.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.3.2 sf_0.9-6     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5         rstudioapi_0.11    magrittr_1.5       units_0.6-7        munsell_0.5.0      tidyselect_1.1.0  
 [7] colorspace_1.4-1   R6_2.5.0           rlang_0.4.7        dplyr_1.0.2        tools_4.0.2        grid_4.0.2        
[13] gtable_0.3.0       KernSmooth_2.23-17 e1071_1.7-3        DBI_1.1.0          withr_2.3.0        ellipsis_0.3.1    
[19] class_7.3-17       tibble_3.0.3       lifecycle_0.2.0    crayon_1.3.4       farver_2.0.3       purrr_0.3.4       
[25] vctrs_0.3.4        glue_1.4.2         compiler_4.0.2     pillar_1.4.6       scales_1.1.1       generics_0.1.0    
[31] classInt_0.4-3     pkgconfig_2.0.3   
```

**Comment by edzer** [comment]
Strange, I get
![x](https://user-images.githubusercontent.com/520851/98238467-b17fcc80-1f66-11eb-9eb0-9936ee407f85.png)
with
```
> sessionInfo()
R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.3.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5         magrittr_1.5       units_0.6-7        tidyselect_1.1.0  
 [5] munsell_0.5.0      colorspace_1.4-1   R6_2.5.0           rlang_0.4.8       
 [9] dplyr_1.0.2        tools_4.0.3        grid_4.0.3         gtable_0.3.0      
[13] KernSmooth_2.23-18 e1071_1.7-4        DBI_1.1.0          withr_2.3.0       
[17] ellipsis_0.3.1     class_7.3-17       tibble_3.0.4       lifecycle_0.2.0   
[21] crayon_1.3.4       sf_0.9-7           farver_2.0.3       purrr_0.3.4       
[25] vctrs_0.3.4        glue_1.4.2         compiler_4.0.3     pillar_1.4.6      
[29] generics_0.1.0     scales_1.1.1       classInt_0.4-3     pkgconfig_2.0.3   
```

**Comment by jonasViehweger** [comment]
Huh, I guess I'll try updating to R 4.0.3, otherwise maybe a Windows issue?

**Comment by jonasViehweger** [comment]
I've just also tested on my installation of Ubuntu where I also don't get the issue.

Maybe someone else can reproduce the issue on Windows?

**Comment by rsbivand** [comment]
R 4.0.2, Windows 10, current ggplot2 and sf:

![test](https://user-images.githubusercontent.com/10198404/98243622-7d100e80-1f6e-11eb-889b-1a87a6ec1cd4.png)

**Comment by rsbivand** [comment]
Same problem with R 4.0.3, Windows 10, ggplot2 3.3.2.

**Comment by edzer** [comment]
This happens in `sf::st_graticule`; the numbers go through `pretty`, `abs`, then `paste0`. On ubuntu this works well, but `paste0` is not instructed to limit the number of digits, and the numbers cannot be represented exactly:
```
> format(abs(pretty(c(8.6847,8.6851))), digits=20)
[1] "8.6846999999999994202" "8.6847999999999991871" "8.6848999999999989541"
[4] "8.6850000000000004974" "8.6851000000000002643"
```
So I guess a `format(x, digits=10)` should happen before passing to `paste0`; see https://xkcd.com/2170/

**Comment by rsbivand** [comment]
But why is Windows different? Anyone have MacOS?

**Comment by edzer** [comment]
Would be great if someone could check this PR on windows.

**Comment by rsbivand** [comment]
How do I install from a github PR?

**Comment by rsbivand** [comment]
With PR 1527, Windows 10:
![test_1527](https://user-images.githubusercontent.com/10198404/98252567-653e8780-1f7a-11eb-9bbc-043b281cce5d.png)

**Comment by edzer** [comment]
Thanks! merged into master.

**Comment by jonasViehweger** [comment]
Perfect, thanks for sorting this out so quickly! I think I can close this issue now.
